File: frontend-server/defaults/counter-test.js
module.exports = `
package main

import (
\t"github.com/orbs-network/orbs-contract-sdk/go/testing/unit"
\t"github.com/stretchr/testify/require"
\t"testing"
)

func TestCounterInitialStateIsZero(t *testing.T) {
\tunit.InSystemScope(nil, nil, func(mockery unit.Mockery) {
\t\trequire.EqualValues(t, 0, get())
\t})
}

func TestCounterStateMutatesAfterAdd(t *testing.T) {
\tunit.InSystemScope(nil, nil, func(mockery unit.Mockery) {
\t\tadd(17)
\t\trequire.EqualValues(t, 17, get())
\t})
}
`

File: goebbels/contracts/counter/counter_test.go
package main

import (
	"github.com/orbs-network/orbs-contract-sdk/go/testing/unit"
	"github.com/stretchr/testify/require"
	"testing"
)

func TestCounterInitialStateIsZero(t *testing.T) {
	unit.InSystemScope(nil, nil, func(mockery unit.Mockery) {
		require.EqualValues(t, 0, get())
	})
}

func TestCounterStateMutatesAfterAdd(t *testing.T) {
	unit.InSystemScope(nil, nil, func(mockery unit.Mockery) {
		add(17)
		require.EqualValues(t, 17, get())
	})
}
